app/helpers/source_control_helper/github_client.rb

```diff
@@ -0,0 +1,82 @@
+module SourceControlHelper
+  # SourceControlHelper::GithubClient Client for GitHub requests
+  class GithubClient
+    attr_reader :server_type
+
+    def initialize
+      Rails.logger.info 'Generating new connection to Source Control'
+
+      github_url = URI::HTTPS.build(host: Rails.application.secrets.github_server, path: "/api/#{Rails.application.secrets.github_version}")
+
+      opts = {
+        api_endpoint:       github_url.to_s,
+        connection_options: { ssl: { verify: Rails.application.secrets.github_verify } },
+        client_id:          Rails.application.secrets.oauth_github_id,
+        client_secret:      Rails.application.secrets.oauth_github_secret,
+        scope:              'user:email'
+      }
+
+      @github_access ||= Octokit::Client.new opts
+
+      @server_type = 'GitHub'
+    end
+
+    ##
+    # Accepts a code and return a token hash
+    # @param code [String]
+    # @return [String]
+    #
+
+    def exchange_code_for_token!(code)
+      github_token = @github_access.exchange_code_for_token(code)
+      @github_access.access_token = github_token[:access_token] unless github_token[:error] || github_token.nil?
+      github_token
+    end
+
+    #
+    # Returns the user
+    # @return user [SourceControlUser]
+    #
+    def user
+      @github_access.user if @github_access.user_authenticated?
+    end
+
+    #
+    # Returns metadata from the repo or nil
+    # @param full_name [String] Full name of repo
+    # @return [metadata_raw, metadata_json]
+
+    def metadata(full_name)
```

Comment:
This should be abstracted to the base class

Reply:
we can move later @chargio, there is things of GitHub client like Octokit. We shouldn't abstract to the base class specific things